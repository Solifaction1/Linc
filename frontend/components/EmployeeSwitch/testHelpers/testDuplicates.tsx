import { Matcher, SelectorMatcherOptions } from '@testing-library/react'

const testDuplicates = (
	container: HTMLElement,
	getByText: (
		id: Matcher,
		options?: SelectorMatcherOptions | undefined
	) => HTMLElement
) => {
	expect(
		container.getElementsByClassName('employeeSwitch className')[0]
	).toBeVisible()

	expect(
		container.getElementsByClassName('employeeLink activeLang')[0]
	).toBeVisible()
	expect(getByText('title', { selector: 'button' })).toBeVisible()
	expect(container.querySelector('svg')).toBeVisible()
	expect(container.querySelector('path')).toBeVisible()

	expect(container.getElementsByClassName('dropDown')[0]).toBeVisible()
}

export default testDuplicates
