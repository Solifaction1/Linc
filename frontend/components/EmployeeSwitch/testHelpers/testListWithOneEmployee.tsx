import { Matcher, SelectorMatcherOptions } from '@testing-library/react'

const testListWithOneEmployee = (
	container: HTMLElement,
	getByText: (
		id: Matcher,
		options?: SelectorMatcherOptions | undefined
	) => HTMLElement,
	urlListIndex: number,
	urlListWrapperIndex: number,
	text: string
) => {
	expect(
		container.getElementsByClassName('urlList')[urlListIndex]
	).toBeVisible()
	const firstUrlListWrapper =
		container.getElementsByClassName('urlListWrapper')[urlListWrapperIndex]
	expect(firstUrlListWrapper).toBeVisible()
	expect(
		firstUrlListWrapper.contains(getByText(text, { selector: 'a' }))
	).toBeTruthy()
}

export default testListWithOneEmployee
