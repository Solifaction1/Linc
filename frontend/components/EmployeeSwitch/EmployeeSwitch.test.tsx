/* eslint-disable max-lines */
import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import EmployeeSwitch from './EmployeeSwitch'
import {
	fiveEmployees,
	fourEmployees,
	oneEmployee,
	parent,
	sevenEmployees,
	sixEmployees,
	threeEmployees,
	twoEmployees,
} from './EmployeeSwitch.config'
import testDuplicates from './testHelpers/testDuplicates'

import testListWithOneEmployee from './testHelpers/testListWithOneEmployee'
import testFirstListWithThreeEmployees from './testHelpers/testFirstListWithThreeEmployees'
import testListWithTwoEmployees from './testHelpers/testListWithTwoEmployees'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('EmployeeSwitch', () => {
	it('renders a EmployeeSwitch with one employee', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: oneEmployee }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testListWithOneEmployee(container, getByText, 0, 0, 'aFirstName aName')

		expect(container.getElementsByClassName('urlList')[1]).toBeVisible()
		expect(
			container.getElementsByClassName('urlListWrapper')[1]
		).toBeUndefined()

		expect(container.getElementsByClassName('urlList')[2]).toBeVisible()
		expect(
			container.getElementsByClassName('urlListWrapper')[2]
		).toBeUndefined()
	})

	it('renders a EmployeeSwitch with two employees', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: twoEmployees }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testListWithOneEmployee(container, getByText, 0, 0, 'aFirstName aName')
		testListWithOneEmployee(container, getByText, 1, 1, 'bFirstName bName')

		expect(container.getElementsByClassName('urlList')[2]).toBeVisible()
		expect(
			container.getElementsByClassName('urlListWrapper')[2]
		).toBeUndefined()
	})

	it('renders a EmployeeSwitch with three employees', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: threeEmployees }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testListWithOneEmployee(container, getByText, 0, 0, 'aFirstName aName')
		testListWithOneEmployee(container, getByText, 1, 1, 'bFirstName bName')
		testListWithOneEmployee(container, getByText, 2, 2, 'cFirstName cName')
	})

	it('renders a EmployeeSwitch with four employees', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: fourEmployees }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testListWithTwoEmployees(
			container,
			getByText,
			0,
			0,
			'aFirstName aName',
			1,
			'bFirstName bName'
		)
		testListWithOneEmployee(container, getByText, 1, 2, 'cFirstName cName')
		testListWithOneEmployee(container, getByText, 2, 3, 'dFirstName dName')
	})

	it('renders a EmployeeSwitch with five employees', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: fiveEmployees }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testListWithTwoEmployees(
			container,
			getByText,
			0,
			0,
			'aFirstName aName',
			1,
			'bFirstName bName'
		)
		testListWithTwoEmployees(
			container,
			getByText,
			1,
			2,
			'cFirstName cName',
			3,
			'dFirstName dName'
		)
		testListWithOneEmployee(container, getByText, 2, 4, 'eFirstName eName')
	})

	it('renders a EmployeeSwitch with six employees', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: sixEmployees }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testListWithTwoEmployees(
			container,
			getByText,
			0,
			0,
			'aFirstName aName',
			1,
			'bFirstName bName'
		)
		testListWithTwoEmployees(
			container,
			getByText,
			1,
			2,
			'cFirstName cName',
			3,
			'dFirstName dName'
		)
		testListWithTwoEmployees(
			container,
			getByText,
			2,
			4,
			'eFirstName eName',
			5,
			'fFirstName fName'
		)
	})

	it('renders a EmployeeSwitch with seven employees', () => {
		const { container, getByText } = render(
			<EmployeeSwitch
				parent={parent}
				employeeData={{ data: sevenEmployees }}
				className="className"
			/>
		)

		expect(container).toMatchSnapshot()

		testDuplicates(container, getByText)

		testFirstListWithThreeEmployees(container, getByText)
		testListWithTwoEmployees(
			container,
			getByText,
			1,
			3,
			'dFirstName dName',
			4,
			'eFirstName eName'
		)
		testListWithTwoEmployees(
			container,
			getByText,
			2,
			5,
			'fFirstName fName',
			6,
			'gFirstName gName'
		)
	})
})
