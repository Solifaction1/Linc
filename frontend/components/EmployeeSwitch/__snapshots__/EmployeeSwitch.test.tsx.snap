// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeSwitch renders a EmployeeSwitch with five employees 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            bFirstName
             
            bName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            cFirstName
             
            cName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            dFirstName
             
            dName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            eFirstName
             
            eName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmployeeSwitch renders a EmployeeSwitch with four employees 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            bFirstName
             
            bName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            cFirstName
             
            cName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            dFirstName
             
            dName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmployeeSwitch renders a EmployeeSwitch with one employee 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      />
      <div
        class="urlList"
      />
    </div>
  </div>
</div>
`;

exports[`EmployeeSwitch renders a EmployeeSwitch with seven employees 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            bFirstName
             
            bName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            cFirstName
             
            cName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            dFirstName
             
            dName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            eFirstName
             
            eName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            fFirstName
             
            fName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            gFirstName
             
            gName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmployeeSwitch renders a EmployeeSwitch with six employees 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            bFirstName
             
            bName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            cFirstName
             
            cName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            dFirstName
             
            dName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            eFirstName
             
            eName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            fFirstName
             
            fName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmployeeSwitch renders a EmployeeSwitch with three employees 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            bFirstName
             
            bName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            cFirstName
             
            cName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmployeeSwitch renders a EmployeeSwitch with two employees 1`] = `
<div>
  <div
    class="employeeSwitch className"
  >
    <button
      class="employeeLink activeLang"
      type="button"
    >
      title
      <svg
        class="false arrow"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
        />
      </svg>
    </button>
  </div>
  <div
    class="dropDown"
    style="height: 0px;"
  >
    <div
      class="urlWrapper"
    >
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            aFirstName
             
            aName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      >
        <div
          class="urlListWrapper"
        >
          <a
            class="link url cta md false"
            href="/employee/undefined"
          >
            bFirstName
             
            bName
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="urlList"
      />
    </div>
  </div>
</div>
`;
